core/src/main/scala/no/nrk/bigquery/BigQueryClient.scala

```diff
@@ -500,27 +500,24 @@ class BigQueryClient[F[_]](
       .flatMap(id => BQMetrics(metricOps, jobId)(loggedJob(id)))
   }
 
-  def getTable(
-      tableId: BQTableId,
-      tableOptions: TableOption*
+  private def getTableImpl(
+      tableId: BQTableId
   ): F[Option[Table]] =
     F.interruptible(
-      Option(bigQuery.getTable(tableId.underlying, tableOptions: _*))
+      Option(bigQuery.getTable(tableId.underlying))
         .filter(_.exists())
     )
 
-  def getTableLike(tableId: BQTableId, tableOptions: TableOption*): F[Option[BQTableDef[Any]]] =
-    OptionT(getTable(tableId, tableOptions: _*)).mapFilter(t => SchemaHelper.fromTable(t).toOption).value
+  private[bigquery] def getExistingTableImpl(tableId: BQTableId): F[Option[ExistingTable]] =
+    OptionT(getTableImpl(tableId))
+      .mapFilter(t => TableHelper.fromTable(t).toOption.map(converted => ExistingTable(converted, t)))
```

Comment:
We should probably fail the F here with the message from the either returned by fromTable